### PR TITLE
feat: Hardhat + ethers/waffle

### PR DESCRIPTION
**Pull request by gakonst**
As title, migrates everything to hardhat/ethers/waffle

### CODE REVIEW

**Comment by alcueca** [comment]
Cool, let me play with it a bit before I merge. Thanks!

File: test/031_delegable.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  })
+
+  it('adds delegates', async () => {
+    await delegable.addDelegate(other)
+    expect(await delegable.delegated(owner, other)).to.be.true
+  })
+
+  describe('with delegates', async () => {
+    beforeEach(async () => {
+      // await delegable.addDelegate(other) <-- Keeps state from 'adds delegates'
```
Comment (alcueca) [inline test/031_delegable.ts]: Why does this not revert the state between tests?

File: package.json [hunk 2]
```diff
@@ -19,41 +19,32 @@
...
-    "coverage": "buidler coverage --network coverage --temp build --testfiles 'test/*.ts'",
+    "build": "hardhat compile",
+    "test": "hardhat test",
+    "coverage": "hardhat coverage",
     "lint": "prettier ./test/**/*.ts --check",
     "lint:ts": "prettier ./test/**/*.ts --write",
     "lint:sol": "solhint -f table contracts/**/*.sol"
   },
-  "dependencies": {
-    "@openzeppelin/contracts": "^3.1.0"
```
Comment (alcueca) [inline package.json:33]: ERC20Permit inherits from the OZ's ERC20.sol. If we remove this, users of the npm package will have to install openzeppelin-contracts themselves, right?

Comment (gakonst) [reply package.json:33]: Yes

File: test/031_delegable.ts (hunk 1 above)
Comment (gakonst) [reply test/031_delegable.ts]: You need to add a `beforeEach` on L16 instead of `before`

**Pull request merged**